jobs/sql/extractions/all_encounters_staging.sql

```diff
@@ -14,49 +14,54 @@ user_entered        varchar(255),
 
-Insert into temp_all_encounters (
-encounter_id, encounter_datetime, encounter_type_name, encounter_location, patient_id, visit_id, user_entered, entered_datetime, voided)
-select encounter_id, encounter_datetime, encounter_type_name_from_id(encounter_type), location_name(location_id), patient_id, visit_id, person_name_of_user(creator), date_created, voided 
+insert into temp_all_encounters (
+encounter_id, encounter_datetime, encounter_type_name, encounter_location, patient_id, visit_id, user_entered, entered_datetime, voided, combined_date_changed)
+select encounter_id, encounter_datetime, encounter_type_name_from_id(encounter_type), location_name(location_id), patient_id, visit_id, person_name_of_user(creator), date_created, voided,
+CASE 
+	when ifnull(date_created,'1000-01-01') >= ifnull(date_changed,'1000-01-01') and ifnull(date_created,'1000-01-01') >= ifnull(date_voided,'1000-01-01') then date_created
+	when ifnull(date_changed,'1000-01-01') >= ifnull(date_created,'1000-01-01') and ifnull(date_changed,'1000-01-01') >= ifnull(date_voided,'1000-01-01') then date_changed
+	else date_voided
+END 
 from encounter 
-where (date_created > @lastloadtime or 
-	date_changed > @lastloadtime or 
-	date_voided > @lastloadtime or 
-	@lastloadtime is null);
+where (ifnull(date_created,'1000-01-01') > @last_loaded_max_time or 
+	 ifnull(date_changed,'1000-01-01') > @last_loaded_max_time or 
+	 ifnull(date_voided,'1000-01-01') > @last_loaded_max_time or 
+	@last_loaded_max_time is null)
+order by 
+CASE 
+	when ifnull(date_created,'1000-01-01') >= ifnull(date_changed,'1000-01-01') and ifnull(date_created,'1000-01-01') >= ifnull(date_voided,'1000-01-01') then date_created
+	when ifnull(date_changed,'1000-01-01') >= ifnull(date_created,'1000-01-01') and ifnull(date_changed,'1000-01-01') >= ifnull(date_voided,'1000-01-01') then date_changed
+	else date_voided
+END asc	
+ limit 5000 -- <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< change
+;
 
--- update last loaded time for encounter domain
-delete from petl_load_times 
-where loaded_domain = 'encounter';
-insert into petl_load_times(loaded_domain, last_loaded_time)
-values ('encounter',now());
+select now() into @current_loaded_time;
+select max(combined_date_changed) into @loaded_max_time from temp_all_encounters;
+select min(combined_date_changed) into @loaded_min_time from temp_all_encounters;
+select count(*) into @number_updated_rows from temp_all_encounters;
+
+-- insert ALL rows with the max load time (ignoring duplicates) to ensure that all rows with that datetime are loaded
+insert ignore into temp_all_encounters (
+encounter_id, encounter_datetime, encounter_type_name, encounter_location, patient_id, visit_id, user_entered, entered_datetime, voided)
+select encounter_id, encounter_datetime, encounter_type_name_from_id(encounter_type), location_name(location_id), patient_id, visit_id, person_name_of_user(creator), date_created, voided 
+from encounter
+where encounter_datetime = @max_loaded_time
```

Comment:
No, this is incorrect, this has nothing to do with the encounter_datetime.  What this is really trying to do is capture any residual encounters with exactly the same combined_date_changed as loaded_max_time but which were not originally included due to the "limit" expression.  So you basically need to do exactly the same insert as you do above, but using `= @last_loaded_max_time` instead of `>= @last_loaded_max_time` and removing the limit.  Right?